holoviews/operation/datashader.py

```diff
@@ -307,7 +334,12 @@ def _process(self, element, key=None):
 
         with warnings.catch_warnings():
             warnings.filterwarnings('ignore', r'invalid value encountered in true_divide')
-            img = tf.shade(array, **shade_opts)
+            if np.isnan(array.data).all():
```

Comment:
Seems like it might be a little inefficient to compute this predicate on large arrays but I think it is okay for now. No need to optimize anything just yet.